### PR TITLE
feat: resource monitoring

### PR DESCRIPTION
**Pull request by cdummett**
### Description
Adds CPU Utilisation and Virtual Memory usage to the `Snitch`. Supporting plots added as well.

example

CPU monitoring supported on Darwin, Windows & Linux. Memory monitoring only supported on Darwin & Linux.

### Testing
Passing all tests locally.

### Breaking Changes
None

### Closes
None

### CODE REVIEW

File: vega_sim/scenario/common/agents.py [hunk 1]
```diff
@@ -2756,11 +2774,66 @@ def step(self, vega_state: VegaState):
...
+                    else 0,
+                )
+            )
         if self.additional_state_fn is not None:
             self.additional_states.append(
                 self.additional_state_fn(self.vega, self.agents)
             )
 
+    def _find_processes(self):
+        self.process_map.keys()
```
Comment (TomMcL) [inline vega_sim/scenario/common/agents.py]: ?

File: vega_sim/scenario/common/agents.py [hunk 2]
```diff
@@ -2756,11 +2774,66 @@ def step(self, vega_state: VegaState):
...
+        for p in psutil.process_iter():
+            if (p.name() in self.process_map.keys()) and (
+                p.status() in ["running", "sleeping"]
+            ):
```
Comment (TomMcL) [inline vega_sim/scenario/common/agents.py]: I wonder whether there's something more robust we can do for this. This is currently going to break if someone ever runs two processes/there are leftover processes. But maybe that just has to be a caveat emptor for now.

Comment (TomMcL) [reply vega_sim/scenario/common/agents.py]: Also I think we'd be better having the process map start empty then have the keys specified here directly, slightly clearer what it's for and avoids a case of `None`s in the dictionary being left over

Comment (cdummett) [reply vega_sim/scenario/common/agents.py]: We could use the `cmdline` property to filter out any processes not started by a `VegaServiceNull` object?

Any other left-over "vega" and "data-node" processes started by other `VegaServiceNull` objects **_should_** be either `stopped` / `zombie` / `killed` by this point.

But yeah if we ever go for a multi-node setup in market-sim we would need to expand this to monitor all nodes.

**Review by TomMcL — APPROVED** [review]
lgtm

**Pull request merged**